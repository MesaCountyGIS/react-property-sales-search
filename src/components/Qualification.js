import React, { Component } from 'react';
import Dropdown from './Dropdown';

const qualificationContent = [
     {type: 'item', value: 'Qualified', label: 'Qualified' },
     {type: 'item', value: 'Unqualified', label: 'Unqualified' },
     {type: 'item', value: 'Qualified and Unqualified', label: 'Qualified and Unqualified' },
]

const qualificationStyle = {
    display: 'block',
    border: '1px solid #ccc',
    borderRadius: '4px',
    padding: '0.5em',
    marginBottom: '1em',
    top: '0',
    height: '2.8em',
    width: '75%',
    boxShadow: '1px 1px 5px #888888'
}
const qualificationLabelStyle = {
    display: 'block',
    fontSize: '0.8em',
    fontWeight: 'bold',
    color: '#C4C4C4',
    margin: '0 0 0.4em 0',
    textAlign:'left',
    borderBottom: '1px solid #ccc'
}

const customStyles = {
    dropNav: {
        "width":"15.5em",
        "marginLeft":"-0.7em",
        "marginTop":"0.9em"
    },
    ulContainer: {
        "width":"14.5em",
        "paddingLeft":"0 !important"
    }
}

  class Qualification extends Component {
    render() {
      return (
          <div style={qualificationStyle} className="qualification">
              <label style={qualificationLabelStyle}>Sales Qualification Type</label>
              <Dropdown
                  baseclass='qualification'
                  customStyles={customStyles}
                  option={qualificationContent}
                  placeholder='Qualified and Unqualified'
                  value={'Qualified and Unqualified'}
              />
          </div>
      );
    }
  }


export default Qualification;
