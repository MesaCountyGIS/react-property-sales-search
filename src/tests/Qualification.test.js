import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {stubConsoleError} from './utils/test-utils'
import Qualification from '../components/Qualification';

describe("<Qualification />", ()=>{
    stubConsoleError();

    test('renders without crashing', () => {
        const tree = renderer.create(
            <Qualification />
      ).toJSON();

      sinon.assert.notCalled(console.error)
      expect(tree).toMatchSnapshot();
    });
});
