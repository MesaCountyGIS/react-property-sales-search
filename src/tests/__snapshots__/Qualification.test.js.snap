// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Qualification /> renders without crashing 1`] = `
<div
  className="qualification"
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "4px",
      "boxShadow": "1px 1px 5px #888888",
      "display": "block",
      "height": "2.8em",
      "marginBottom": "1em",
      "padding": "0.5em",
      "top": "0",
      "width": "12em",
    }
  }
>
  <label
    style={
      Object {
        "borderBottom": "1px solid #ccc",
        "color": "#C4C4C4",
        "display": "block",
        "fontSize": "0.8em",
        "fontWeight": "bold",
        "margin": "0 0 0.4em 0",
        "textAlign": "left",
      }
    }
  >
    Sales Qualification Type
  </label>
  <div
    className="dropdown qualification"
    style={
      Object {
        "paddingLeft": "0 !important",
        "width": "14.5em",
      }
    }
  >
    Qualified and Unqualified
    <span
      className="dropdown qualification"
    >
      â–¼
    </span>
    <ul
      className="drop-nav display"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": "-0.7em",
          "marginTop": "0.9em",
          "width": "15.5em",
        }
      }
    >
      <li
        className="item"
      >
        Qualified
      </li>
      <li
        className="item"
      >
        Unqualified
      </li>
      <li
        className="item"
      >
        Qualified and Unqualified
      </li>
    </ul>
  </div>
</div>
`;
